WordPress/gutenberg
# Font Library modal: Try to improve checkbox labelling

## What?

This PR is an attempt to address https://github.com/WordPress/gutenberg/issues/58299, and covers the following:

- Reverts the changes to the `CheckboxControl` component added in https://github.com/WordPress/gutenberg/pull/56158
- Tries an alternative labelling solution for the checkboxes in the Font Library modal without needing to adjust the `CheckboxControl` component

## Why?

To improve accessibility and usability of these checkboxes whilst maintaining the functionality of the `CheckboxControl` component.

## How?

The main challenge with the current implementation of the Font Library checkboxes is that the visible label can either be text or an image, depending on if the font face provides a preview image (e.g. an SVG). This logic cannot currently be passed to the `CheckboxControl` component, as the `label` prop only accepts a string (which I agree with).

Also, the older design for the font library checkboxes had the checkboxes to the right of the label, rather than the left. Now the design has been updated to be in line with the `CheckboxControl` component, where the checkbox is to the left of the label. This makes it much easier to use the existing functionality of this component.

The solution in this PR is to provide a label in the `CheckboxControl`, and then hide the label element that the component creates using CSS. The CSS should visibly hide the label while still allowing it to be accessed by screen-readers. This has worked well in my testing so far.

There has also been an `onClick` handler added to the `FontFaceDemo` component, which allows the checkbox to be toggled when this component is clicked. Otherwise, only the checkbox itself can be interacted with to toggle the checkbox value, and I don't think this is good for accessibility or usability.

This is one attempt at an alternative solution to handling these checkbox labels, and I'd love feedback on whether this is a good approach or if we should try alternatives.

Possible other alternatives:
- Use plain text as the label, and move the font demo to a separate element (e.g. a column layout, where the font demo sits to the right of the text label). This would require design input.
- Add `aria-labelledby` attribute to the `CheckboxControl`, and reference the `FontFaceDemo` element as the label
- Allow a custom label functionality to be passed to the `CheckboxControl` component (that doesn't allow the label to be `false`)

## Testing Instructions




1. Ensure you're using a theme that includes at least one bundled font family (or upload a font family as part of this test)
2. Open the Font Library modal
3. Open a font family
4. Toggle a font variant of the selected font family by clicking either the checkbox or the text/image font preview

### Testing Instructions for Keyboard

1. Ensure you're using a theme that includes at least one bundled font family (or upload a font family as part of this test)
2. Open the Font Library modal
3. Open a font family
4. Toggle a font variant of the selected font family by tabbing to the font variant panel and toggling the checkbox. If using a screenreader, the font variant name should be called out.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/1645628/114bb980-1232-4933-9bdb-9ab84ef733b3
